src/main/java/io/getunleash/DefaultUnleash.java

```diff
@@ -50,7 +51,9 @@
     private static FeatureToggleRepository defaultToggleRepository(UnleashConfig unleashConfig) {
         return new FeatureToggleRepository(
                 unleashConfig,
-                new HttpToggleFetcher(unleashConfig),
+                unleashConfig.isUseOkHttpClient()
```

Comment:
Instead of a boolean, it would be nice of unleashConfig to allow providing a fetcher implementation:

```
unleashConfig.toggleFetcher(new OkHttpToggleFetcher(unleashConfig))
```

Though that does mean the the unleash config being passed in is being created when the fetcher is being constructed.

Reply:
To work around the config being created when the instantiation happens, we should make it a factory:

```java
unleashConfig.toggleFetcherFactory(config -> new OkHttpToggleFetcher(config))
```

Reply:
Thanks Clint, another good point :+1: I'll update the PR.